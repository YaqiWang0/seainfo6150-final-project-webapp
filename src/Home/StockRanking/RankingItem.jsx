import React, { useEffect, useState} from "react"
import { useHistory, useLocation } from "react-router-dom";
import style from "./StockRanking.module.css"

const RankingItem = (props) => {
    // TODO: semetic html
    let history = useHistory();
    const openDetail = async (e) => {
        const response = await fetch(`https://financialmodelingprep.com/api/v3/profile/${props.stock.ticker}?apikey=56f622a93b55019c5a0875058fc58f13`);
        const responseJson = await response.json();
        const newsResponse = await fetch(`https://financialmodelingprep.com/api/v3/stock_news?tickers=${props.stock.ticker}&limit=10&apikey=56f622a93b55019c5a0875058fc58f13`);
        const newsResponseJson = await newsResponse.json();
        history.push(
            "/company",
            {
                info: Object.values(responseJson),
                news: newsResponseJson,
                login: props.login
            }
        )
    }

    return (
    <div className={style.ranking_item} onClick={openDetail}>
        <div className={style.name}>
            <span className={style.stock_name}>{props.stock.ticker}</span>
            <span className={style.company_name}>{props.stock.companyName}</span>
        </div>
        <span className={props.isGainer ? style.gainer_price_text : style.loser_price_text}>{`$${props.stock.price}`}</span>
    </div>
    );
}

export default RankingItem