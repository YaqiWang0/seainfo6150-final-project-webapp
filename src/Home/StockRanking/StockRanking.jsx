import React, { useEffect, useState} from "react"
import Gainer from "./Gainer/Gainer";
import Loser from "./Loser/Loser";
import style from "./StockRanking.module.css"

const StockRanking = (props) => {
    return (
        <div className={style.ranking_container}>
            <Gainer data={Object.values(props.gainer)} login = {props.login}/>
            <Loser data={Object.values(props.loser)} login = {props.login}/>
        </div>
    )
}

export default StockRanking