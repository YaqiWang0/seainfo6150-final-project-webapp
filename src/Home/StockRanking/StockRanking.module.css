

.gainer_price_text {
    color: red;
    padding-top: 0.5em;
    font-size: 2em;
    font-weight: bold;
}

.loser_price_text {
    color: green;
    font-size: 2em;
    padding-top: 0.5em;
    font-weight: bold;
}

.name {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 40%;
    height: 100%;
}

.stock_name {
    font-size: large;
    color: white;
}

.company_name {
    color: gray;
    font-family: Serif;
    width: 100%;
    overflow: hidden;
    white-space: nowrap;
    text-align: left;
}

.ranking_item {
    cursor: pointer;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    height: 3em;
    margin: 1vh;
}

.line {
    border: 1px solid white;
    opacity: 50%;
}
@media screen and (min-width: 0px){
    .company_name {
        display: none;
    }

    .ranking_container {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        padding-top: 5%;
        align-items: center;
    }

    .gainer {
        width: 80%;
        border-radius: 10%;
        background-color: black;
        padding: 3vh;
        margin-bottom: 3vh;
    }

    .loser {
        width: 80%;
        border-radius: 10%;
        background-color: black;
        padding: 3vh;
    }

    .name {
        justify-content: center;
    }
}

@media screen and (min-width: 900px) {
    .ranking_container {
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
        padding-top: 5%;
    }

    .gainer {
        width: 40%;
        border-radius: 10%;
        background-color: black;
        padding: 3vh;
        margin-bottom: 0;
    }

    .loser {
        width: 40%;
        border-radius: 10%;
        background-color: black;
        padding: 3vh;
    }

    .company_name {
        display: block;
    }

    .name {
        justify-content: space-between;
    }
}
