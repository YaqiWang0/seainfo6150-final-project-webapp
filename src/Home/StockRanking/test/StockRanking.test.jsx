import React from "react";
import data from "./GainerAndLoserData"
import StockRanking from "../StockRanking";

describe("StockRanking test", () => {
    const login = {
        login: true,
        userId: "12345",
        email: "test@test.com"

    }
    it('when the item is not clicked', function () {
        const {container} = render(<StockRanking gainer = {data.gainer} loser = {data.loser} login = {login}/>);
        expect(container).toMatchSnapshot();
    })
})