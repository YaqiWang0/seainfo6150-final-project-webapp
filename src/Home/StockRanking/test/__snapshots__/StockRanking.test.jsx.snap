// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockRanking test when the item is not clicked 1`] = `
<div>
  <div
    class="ranking_container"
  >
    <div
      class="gainer"
    >
      <div
        class="title"
      >
         Most Gainer Stock Today
      </div>
      <div
        class="list"
      >
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                SNOA
              </span>
              <span
                class="company_name"
              >
                Sonoma Pharmaceuticals Inc
              </span>
            </div>
            <span
              class="gainer_price_text"
            >
              $10.75
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                SLDB
              </span>
              <span
                class="company_name"
              >
                Solid Biosciences Inc
              </span>
            </div>
            <span
              class="gainer_price_text"
            >
              $5.7706
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                EQS
              </span>
              <span
                class="company_name"
              >
                EQUUS Total Return Inc
              </span>
            </div>
            <span
              class="gainer_price_text"
            >
              $1.755
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                NH
              </span>
              <span
                class="company_name"
              >
                NantHealth Inc
              </span>
            </div>
            <span
              class="gainer_price_text"
            >
              $4.18
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                VTVT
              </span>
              <span
                class="company_name"
              >
                vTv Therapeutics Inc
              </span>
            </div>
            <span
              class="gainer_price_text"
            >
              $3.2
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
      </div>
    </div>
    <div
      class="loser"
    >
      <div
        class="title"
      >
         Most Loser Stock Today
      </div>
      <div
        class="list"
      >
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                GLSI
              </span>
              <span
                class="company_name"
              >
                Greenwich LifeSciences, Inc.
              </span>
            </div>
            <span
              class="loser_price_text"
            >
              $46.52
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                SLS
              </span>
              <span
                class="company_name"
              >
                Sellas Life Sciences Group Inc
              </span>
            </div>
            <span
              class="loser_price_text"
            >
              $11.7
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                NBRV
              </span>
              <span
                class="company_name"
              >
                Nabriva Therapeutics PLC
              </span>
            </div>
            <span
              class="loser_price_text"
            >
              $2.455
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                KBLMU
              </span>
              <span
                class="company_name"
              >
                KBL Merger Corp IV
              </span>
            </div>
            <span
              class="loser_price_text"
            >
              $8.38
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
        <div>
          <div
            class="ranking_item"
            role="click"
          >
            <div
              class="name"
            >
              <span
                class="stock_name"
              >
                IPDN
              </span>
              <span
                class="company_name"
              >
                Professional Diversity Network, Inc.
              </span>
            </div>
            <span
              class="loser_price_text"
            >
              $1.495
            </span>
          </div>
          <hr
            class="line"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
